packages/opentelemetry-core/src/platform/node/environment.ts

```diff
@@ -23,15 +22,31 @@ import {
 } from '../../utils/environment';
 
 /**
- * Gets the environment variables
+ * Gets the environment variables.
+ * Used directly ONLY in test files.
+ * Accessing env vars is done by getEnv().
  */
-export function getEnv(): Required<ENVIRONMENT> {
+export function calculateEnv(): Required<ENVIRONMENT> {
```

Comment:
Rather than exporting this function, I'd find a function (presumably "resetEnvCache`) that resets the `calculateEnvCache` would make more sense here. With the ability to reset the cache, `getEnv()` can also be updated once we definitely know the environs are been updated.

Reply:
What do you think about an optional param(presumably `resetCache`) to the `getEnv` function? 
This way there is no need to export `calculateEnv()`

Then in the case of -
1. testing
2. once we know environs are updated 

We can set `resetCache` to true.

Reply:
I don't have a strong opinion on this but I personally don't like a "get" function to make side effects.